net.go

```diff
@@ -19,37 +19,16 @@ const minPortRange = 1
 
 const maxPortRange = 65535
 
-// IPAddrs finds and returns a list of non-loopback IP addresses of the
-// current machine.
-func IPAddrs() (ips []net.IP, err error) {
-	ips = []net.IP{}
-
-	addrs, err := net.InterfaceAddrs()
-	if err != nil {
-		return
-	}
-
-	for _, a := range addrs {
-		if ipnet, ok := a.(*net.IPNet); ok {
-			if ipnet.IP.IsGlobalUnicast() && !ipnet.IP.IsLinkLocalUnicast() {
-				ips = append(ips, ipnet.IP)
-			}
-		}
-	}
-
-	return ips, nil
-}
-
 // IPFilterFunc filters IP addresses.
 //
 // List of IPFilterFuncs is applied with AND logic: all filters should return true
 // for the IP address to be included in the response.
-type IPFilterFunc func(ip net.IP) bool
+type IPFilterFunc func(addr netip.Addr) bool
 
 // IPFilter filters list of IP addresses by applying sequence of filters.
 //
 // If any of the filters returns false, address is skipped.
-func IPFilter(addrs []net.IP, filters ...IPFilterFunc) (result []net.IP) {
+func IPFilter(addrs []netip.Addr, filters ...IPFilterFunc) (result []netip.Addr) {
```

Comment:
don;t we need to update places that use this method, or the passed argument already implements `netip.Addr`?

Reply:
this is going to be a breaking change, the goal is to get rid of `net.IP`

most of Talos is `net.IP`-free, but when we get to this module, we convert from `netip.Addr` to `net.IP` and back, so basically this is path forward for Talos

Reply:
figured out this is unused, so dropped it lol

Reply:
ah I thought I was looking at talos PR :sweat_smile: